tools/testsys/src/run.rs

```diff
@@ -0,0 +1,217 @@
+use crate::aws_resources::AwsK8s;
+use anyhow::{anyhow, ensure, Context, Result};
+use bottlerocket_variant::Variant;
+use clap::Parser;
+use log::{debug, info};
+use model::test_manager::TestManager;
+use model::SecretName;
+use pubsys_config::InfraConfig;
+use serde::Deserialize;
+use serde_plain::derive_fromstr_from_deserialize;
+use std::collections::HashMap;
+use std::fs::File;
+use std::path::PathBuf;
+
+/// Run a set of tests for a given arch and variant
+#[derive(Debug, Parser)]
+pub(crate) struct Run {
+    /// The type of test to run. Options are `quick` and `conformance`.
+    test_flavor: TestType,
+
+    /// The architecture to test. Either x86_64 or aarch64.
+    #[clap(long, env = "BUILDSYS_ARCH")]
+    arch: String,
+
+    /// The variant to test
+    #[clap(long, env = "BUILDSYS_VARIANT")]
+    variant: String,
+
+    /// The path to `Infra.toml`
+    #[clap(long, env = "PUBLISH_INFRA_CONFIG_PATH", parse(from_os_str))]
+    infra_config_path: PathBuf,
+
+    /// The path to `amis.json`
+    #[clap(long, env = "AMI_INPUT")]
+    ami_input: String,
+
+    /// Override for the region the tests should be run in. If none is provided the first region in
+    /// Infra.toml will be used. This is the region that the aws client is created with for testing
+    /// and resource agents.
+    #[clap(long, env = "TESTSYS_TARGET_REGION")]
+    target_region: Option<String>,
+
+    /// The name of the cluster for resource agents (eks resource agent, ecs resource agent). Note:
+    /// This is not the name of the `testsys cluster` this is the name of the cluster that tests
+    /// should be run on. If no cluster name is provided, the bottlerocket cluster
+    /// naming convention `<arch>-<variant>` will be used.
+    #[clap(long, env = "TESTSYS_TARGET_CLUSTER_NAME")]
+    target_cluster_name: Option<String>,
+
+    /// The custom kube conformance image that should be used by sonobuoy. This is only applicable
+    /// for k8s variants. It can be ommitted for non-k8s variants and it can be ommitted to use the
```

Comment:
nit:
```suggestion
    /// for k8s variants. It can be omitted for non-k8s variants and it can be omitted to use the
```

(There's another instance of this typo later too.)